CHANGELOG.md

```diff
@@ -27,9 +27,10 @@ The format is based on [Keep a Changelog].
 - The `IBMQBackend.run()` function now accepts an optional `job_name` parameter.
   If specified, the `job_name` is assigned to the job. This `job_name` can 
   subsequently be used as a filter in the `IBMQBackend.jobs()` function, which
-  now accepts an optional `job_name` parameter. `job_name` does not need to be 
+  now accepts an optional `job_name` parameter (allowing partial matching and
+  use of regular expressions). `job_name` does not need to be
```

Comment:
Self-reminder to clean up a bit the changelog before the release - I mostly edited it during this PR without too much concern about the specific wording so we remember later.